pkg/buildpack/package.go

```diff
@@ -116,18 +117,15 @@ func extractBuildpacks(pkg Package) (mainBP BuildModule, depBPs []BuildModule, e
 					}
 					return rc, nil
 				}
-				seenDiffIDs[diffID] = member
+				processedDiffIDs[diffID] = true
 			}
 
 			b := &openerBlob{
 				opener: openerFunc,
 			}
 
 			var blobOpts []BlobOption
-			if _, ok := flattenedDiffIDs[diffID]; ok {
-				blobOpts = append(blobOpts, Flattened())
-			}
-			if desc.Info().Match(md.ModuleInfo) { // This is the order buildpack of the package
+			if desc.Info().Match(md.ModuleInfo) { // Current module is the order buildpack of the package
 				mainBP = FromBlob(&desc, b, blobOpts...)
```

Comment:
Because we are removing method `ContainsFlattenedModules()`, I think we can also remove the *BlobOption*